### ISSUE TITLE
Is the Training Curves of SCLD on GMM40 Expected?

### ISSUE THREAD

**Issue by minghuix98** [anonymous3141/SCLD#1]
Hi authors,

Thank you so much for your excellent work!

I just reproduced the SCLD on the GMM40 target and obtained the following curves (the yellow one includes the buffer, while the purple one does not). The curves show significant changes during training.

However, the final sampled posteriors appear reasonable and align well with the true samples along some dimensions.

I'm wondering—is this behavior expected?

![Image](https://github.com/user-attachments/assets/6d4f2ce3-b876-4dc3-8ad9-4e89028600db)

![Image](https://github.com/user-attachments/assets/23704d1b-162d-4710-a09b-09d6c9b007a8)

![Image](https://github.com/user-attachments/assets/ee012cf7-e84e-4b90-b3f8-59e729bf0b16)

Below are some library versions:
jax                          0.4.28
jaxlib                       0.4.28+cuda12.cudnn89
jaxopt                       0.8.3
jaxtyping                    0.3.1
jmp                          0.0.4
joblib                       1.4.2
keras                        3.9.2
kiwisolver                   1.4.8
libclang                     18.1.1
lineax                       0.0.7
Markdown                     3.7
markdown-it-py               3.0.0
MarkupSafe                   3.0.2
matplotlib                   3.8.4
mdurl                        0.1.2
mergedeep                    1.3.4
ml_collections               0.1.1
ml-dtypes                    0.3.2
msgpack                      1.1.0
multipledispatch             1.0.0
namex                        0.0.8
nest-asyncio                 1.6.0
numpy                        1.26.4
numpyro                      0.14.0
nvidia-cublas-cu12           12.6.4.1
nvidia-cuda-cupti-cu12       12.8.90
nvidia-cuda-nvcc-cu12        12.8.93
nvidia-cuda-nvrtc-cu12       12.8.93
nvidia-cuda-runtime-cu12     12.8.90
nvidia-cudnn-cu12            8.9.7.29
nvidia-cufft-cu12            11.3.3.83
nvidia-cusolver-cu12         11.7.1.2
nvidia-cusparse-cu12         12.5.8.93
nvidia-nccl-cu12             2.26.2
nvidia-nvjitlink-cu12        12.8.93

**Comment by anonymous3141** [comment]
Hey,

Thanks so much for your interest in the work!! It is nice that the samples look OK, and indeed from experience we did see some sizable oscillation during training. Especially for multimodal tasks like GMM40-50d, the replay buffer is good at reducing this. However we didn't see anything as crazy as what you are showing, so this is quite strange. Most diffusion samplers are quite sensitive to details and hyperparameters (e.g for some diffusion samplers even going from float32 -> float64 makes a difference on some densities), did you change anything in the config? Moreover, whats your hardware you're running on?

**Comment by minghuix98** [comment]
Hi, 

Thanks for your prompt reply!
I don't think I changed anything in the config — I just cloned the repo and ran it.
I'm running it on a Linux server with an NVIDIA A100 GPU, and the CUDA version is 12.4.
By the way, I also ran another method called Underdamped Diffusion Bridges with Applications to Sampling, after discussing with Julius. The loss curves for that method are not oscillating—one of them (for Funnel-10D) is shown below. I can test it on a higher-dimensional case like GMM-50D later.
Thanks for your time!

![Image](https://github.com/user-attachments/assets/33593ce8-98ba-4dd8-b5e7-e8359743367f)

Best,
Minghui

**Comment by anonymous3141** [comment]
Thanks for confirming, I'm not sure whats going on there (and I don't recall the clipping behavior occurring in the loss when I ran GMM40-50d). I'll run my code again when time permits. Glad to hear underdamped bridges is a good alternative though!

**Comment by minghuix98** [comment]
Thank you so much.
This reminds me that I also encountered some abnormalities with DBS when running on my Windows laptop instead of the server.
As shown below, for dbs_ud, the loss suddenly becomes NaN.
So I suspect this might be caused by some underlying changes in JAX (a different version than the one specified in the DBS repo) or some floating-point settings, as you mentioned earlier.
Once again, thank you for your excellent work!

>python run.py algorithm=dbs_ud target=funnel
Step 0: Loss: inf; ELBO -13.77349853515625; Δ lnZ -2.474196434020996; reverse_ESS 0.006936655845493078
Step 100: Loss: 2.9065282344818115; ELBO -2.8589258193969727; Δ lnZ -1.1671838760375977; reverse_ESS 0.030017029494047165
Step 200: Loss: 2.3536646366119385; ELBO -2.451206684112549; Δ lnZ 0.8925685882568359; reverse_ESS 0.0006724722334183753
Step 300: Loss: 2.219909429550171; ELBO -2.1289641857147217; Δ lnZ -1.0763845443725586; reverse_ESS 0.03003741055727005
Step 400: Loss: 2.0121190547943115; ELBO -2.0226738452911377; Δ lnZ -1.0873527526855469; reverse_ESS 0.01684490032494068
Step 500: Loss: nan; ELBO nan; Δ lnZ nan; reverse_ESS nan
Step 600: Loss: nan; ELBO nan; Δ lnZ nan; reverse_ESS nan

My hardware is:
NVIDIA GeForce RTX 4070
NVIDIA-SMI 561.03                 Driver Version: 561.03         CUDA Version: 12.6

Below are some library versions:
jax                          0.4.34
jaxlib                       0.4.34
jaxopt                       0.8.3
jaxtyping                    0.3.1
jmp                          0.0.4
joblib                       1.4.2
keras                        3.9.2
kiwisolver                   1.4.8
libclang                     18.1.1
lineax                       0.0.7
Markdown                     3.7
markdown-it-py               3.0.0
MarkupSafe                   3.0.2
matplotlib                   3.8.4
mdurl                        0.1.2
mergedeep                    1.3.4
ml_collections               0.1.1
ml-dtypes                    0.3.2
msgpack                      1.1.0
multipledispatch             1.0.0
namex                        0.0.8
nest-asyncio                 1.6.0
numpy                        1.26.4
numpyro                      0.15.0
nvidia-cublas-cu12           12.8.4.1
nvidia-cuda-nvrtc-cu12       12.8.93
nvidia-cudnn-cu12            8.9.7.29
